packages/x-components/src/x-modules/facets/components/filters/__tests__/hierarchical-filter.spec.ts

```diff
@@ -128,360 +102,308 @@ describe('testing `HierarchicalFilter` component', () => {
-      const { wrapper, filter, updateFilter } = renderHierarchicalFilter({
-        filter: getHierarchicalFilterStub({
-          id: 'root',
-          selected: false,
-          label: 'Root: Partially Selected',
-          children: [
-            getHierarchicalFilterStub({
-              id: 'child-0',
-              selected: true,
-              label: 'Child 0: Partially-Selected',
-              children: [
-                getHierarchicalFilterStub({
-                  selected: false,
-                  id: 'grand-child-0',
-                  label: 'GrandChild 0: Unselected'
-                }),
-                getHierarchicalFilterStub({
-                  id: 'grand-child-1',
-                  selected: true,
-                  label: 'GrandChild 1: Selected'
-                })
-              ]
-            })
-          ]
-        }),
+    it('exposes proper css classes and attributes in the default slot to children', () => {
+      const { setFilter, getFiltersWrappers } = renderHierarchicalFilter({
         template: `
-          <HierarchicalFilter
-            :filter="filter"
-            v-slot="{ filter, clickFilter, cssClasses, isDisabled }"
-          >
-            <button
-              :data-test="filter.id"
-              :class="cssClasses"
-              @click="clickFilter"
-              :disabled="isDisabled"
-            >
-                {{ filter.label }}
-            </button>
-          </HierarchicalFilter>
-        `
+           <HierarchicalFilter
+             :filter="filter"
+             v-slot="{ filter, clickFilter, cssClasses, isDisabled }"
+             >
+               <button
+               data-test="filter"
+               :class="cssClasses"
+               @click="clickFilter"
+               :disabled="isDisabled"
+               >
+                 {{ filter.label }}
+               </button>
+           </HierarchicalFilter>
+     `
       });
-      const childrenFilterWrapper = wrapper.find(
-        getDataTestSelector(filter.children![0].id.toString())
-      );
-
-      expect(childrenFilterWrapper.attributes()).not.toHaveProperty('disabled');
-      expect(childrenFilterWrapper.classes()).toHaveLength(6);
-      expect(childrenFilterWrapper.classes()).toEqual(
-        expect.arrayContaining([
-          'x-filter',
-          'x-filter--is-selected',
-          'x-filter--is-partially-selected',
-          'x-hierarchical-filter',
-          'x-hierarchical-filter--is-selected',
-          'x-hierarchical-filter--is-partially-selected'
-        ])
-      );
-
-      await updateFilter({
-        children: [
-          Object.assign(filter.children![0], { selected: false, totalResults: 0, children: [] })
-        ]
+      const filtersWrappers = getFiltersWrappers();
+
+      expect(filtersWrappers.length).toBeGreaterThan(1);
+
+      filtersWrappers.wrappers.forEach(async filterWrapper => {
+        const filter = (filterWrapper.vm as unknown as any).filter;
```

Comment:
Why so many `as unknown as any` ? just with a single `as any` is enough to completely lose types 🤣, or better, you can change the type of the wrappers of the `HierarchicalFilterAPI` to expose the `filter` property in the `vm` property. For example: `Wrapper<Vue & { filter: HierarchicalFilterModel }>`. In any case, I won't fight for these tests, they are not the thing to focus in in this PR.

Reply:
done